src/pages/students/forms/planDataForm/PlanData.tsx

```diff
@@ -0,0 +1,81 @@
+import { useForm } from "react-hook-form";
+import { ErrorMessage} from "../../../../components/error_message/ErrorMessage";
+import { planDataScheme } from "./PlanData.scheme";
+import { shiftRegistrationCalendarData } from "../../../../components/shiftRegistrationCalendar/ShiftRegistrationCalendarData";
+import { MainContainer,
+        Label, 
+        Select, 
+        FormContainer, 
+        Button, 
+        ButtonsContainer} 
+        from "./PlanData.styles";
+import * as yup from "yup";
+import { yupResolver } from "@hookform/resolvers/yup";
+import ShiftRegistrationCalendar from "../../../../components/shiftRegistrationCalendar/ShiftRegistrationCalendar";
+import ShiftDetail from "../../../../components/shiftsDetail/ShiftDetail";
+import { useShiftHandler } from "../../../../components/shiftRegistrationCalendar/UseShiftHandler";
+import useAuthentication from "../../../../hooks/useAuthentication";
+import { useGetUserPlansQuery } from "../../../../app/services/UserService";
+import type { PlanDataProps } from "../../create-student/CreateStudent";
+
+interface FormProp {
+  createStudentCall: (data: PlanDataProps) => void;
+    planData: PlanDataProps | undefined;
+    stepBack:()=>void;
+}
+function PlanData({
+  createStudentCall,
+  planData,
+  stepBack
+}: FormProp) {
```

Comment:
Y aca ahora vamos a usar esa misma interfaz


```suggestion
function PlanData({
  onNext,
  onBack,
  data
}: FormProp<PlanDataProps>) {
```

Fijate lo importante del generico, estamos permitiendo que los componentes implementen la interfaz que define las acciones en comun, pero cada formulario puede declarar el tipo de dato que utiliza, entonces a la hora de enviar la informacion desde el `CreateStudent` vamos a enviar solo lo que el formulario desee